crates/boundless-market/build.rs

```diff
@@ -222,11 +236,44 @@ fn copy_artifacts() {
                 .unwrap()
                 .trim_start_matches("0x");
 
-            // Write to new file with .hex extension
-            let dest_file = dest_path.join(format!("{contract}.hex"));
-            let mut file = File::create(dest_file).unwrap();
-            file.write_all(bytecode.as_bytes()).unwrap();
+            // Append the contract definition with embedded bytecode
+            rust_content.push_str(&format!(
+                r#"alloy::sol! {{
+    #[sol(rpc, bytecode = "{}")]
+    contract {} {{
+        {}
+    }}
+}}"#,
+                bytecode,
+                contract,
+                get_interfaces(contract)
+            ));
+
+            // Only add newline between contracts, not after the last one
+            if contract != *ARTIFACT_TARGET_CONTRACTS.last().unwrap() {
+                rust_content.push_str("\n\n");
+            }
+        }
+    }
+    rust_content.push('\n');
+    let dest_path = Path::new(&manifest_dir).join("src/contracts/bytecode.rs");
+    fs::write(dest_path, rust_content).unwrap();
+}
+
+// Helper function to define interfaces for each contract
+fn get_interfaces(contract: &str) -> &str {
+    match contract {
+        "RiscZeroMockVerifier" => "constructor(bytes4 selector) {}",
+        "RiscZeroSetVerifier" => {
+            "constructor(address verifier, bytes32 imageId, string memory imageUrl) {}"
+        }
+        "BoundlessMarket" => {
+            r#"constructor(address verifier, bytes32 assessorId, address stakeTokenContract) {}
+            function initialize(address initialOwner, string calldata imageUrl) {}"#
         }
+        "ERC1967Proxy" => "constructor(address implementation, bytes memory data) payable {}",
+        "HitPoints" => "constructor(address initialOwner) payable {}",
+        _ => "",
```

Comment:
These need to match the actual constructors and must be kept consistent. As an alternative to the entire bytecode.rs file we could as this point also copy the entire compiled json file (and keep it checked in). Then we would have interface and bytecode and a default.

Reply:
Unfortunately checking in the json does not work as they are too susceptible to changes during different compilations/environments

Reply:
Maybe it is not worth for now, as when https://github.com/alloy-rs/core/pull/898 gets into a release we could revert to using .hex files